crates/bitwarden-error-macro/src/flat/attribute.rs

```diff
@@ -0,0 +1,117 @@
+use quote::quote;
+use syn::Data;
+
+pub(crate) fn bitwarden_error_flat(
+    input: &syn::DeriveInput,
+    type_identifier: &proc_macro2::Ident,
+    export_as_identifier: &proc_macro2::Ident,
+) -> proc_macro::TokenStream {
+    match &input.data {
+        Data::Enum(data) => {
+            let variant_names = data.variants.iter().map(|variant| &variant.ident);
+
+            let match_arms = data.variants.iter().map(|variant| match variant.fields {
+                syn::Fields::Unit => {
+                    let variant_ident = &variant.ident;
+                    let variant_name = format!("{}", variant_ident);
+                    quote! {
+                        #type_identifier::#variant_ident => #variant_name
+                    }
+                }
+                syn::Fields::Named(_) => {
+                    let variant_ident = &variant.ident;
+                    let variant_name = format!("{}", variant_ident);
+                    quote! {
+                        #type_identifier::#variant_ident { .. } => #variant_name
+                    }
+                }
+                syn::Fields::Unnamed(_) => {
+                    let variant_ident = &variant.ident;
+                    let variant_name = format!("{}", variant_ident);
+                    quote! {
+                        #type_identifier::#variant_ident(..) => #variant_name
+                    }
+                }
+            });
+
+            let wasm = cfg!(feature = "wasm").then(|| {
+                flat_error_wasm(
+                    type_identifier,
+                    export_as_identifier,
+                    &variant_names.collect::<Vec<_>>(),
+                )
+            });
+
+            quote! {
+                #input
+                #wasm
+
+                #[automatically_derived]
+                impl FlatError for #type_identifier {
```

Comment:
When referring to types in macro generated code, we want to always use the fully qualified names, otherwise we're requiring users to import the correct type. If you're importing`prelude::*` that's not an issue, but nothing keeps the user from directly importing just the macro.
```suggestion
                impl ::bitwarden_error::prelude::FlatError for #type_identifier {
```

Note that we also use `::` in front of the crate name, this is to avoid collisions with user defined modules using the same name.

Reply:
I did this for types that we depend on (e.g. `wasm-bindgen`) but never considered our own types, good catch!